File: solutions/2026-06-25/solution.py
# Valid Parentheses
# Difficulty: Medium
# URL: https://leetcode.com/problems/valid-parentheses/

def validParentheses(s: str) -> bool:
  stack = []
  bracket_map = {')': '(', '}': '{', ']': '['}
  for c in s:
    if c in bracket_map.values():
      stack.append(c)
    elif c in bracket_map.keys():
      if not stack or bracket_map[c] != stack.pop():
        return False
  return not stack

File: solutions/2026-06-25/solution.java
// Valid Parentheses
// Difficulty: Medium
// URL: https://leetcode.com/problems/valid-parentheses/

public boolean validParentheses(String s) {
  Stack<Character> stack = new Stack<>();
  Map<Character, Character> bracketMap = new HashMap<>();
  bracketMap.put(')', '(');
  bracketMap.put('}', '{');
  bracketMap.put(']', '[');
  for (char c : s.toCharArray()) {
    if (bracketMap.containsValue(c)) {
      stack.push(c);
    } else if (bracketMap.containsKey(c)) {
      if (stack.isEmpty() || bracketMap.get(c) != stack.pop()) {
        return false;
      }
    }
  }
  return stack.isEmpty();
}